src/Sentry/Protocol/Envelope.cs

```diff
@@ -14,6 +15,15 @@ public Envelope(EnvelopeHeaderCollection headers, EnvelopeItemCollection items)
             Items = items;
         }
 
+        /// <summary>
+        /// Attempts to extract the value of "sentry_id" header if it's present.
+        /// </summary>
+        public SentryId? TryGetEventId() =>
+            Headers.KeyValues.TryGetValue("event_id", out var value) &&
+            value is string valueString
+                ? new SentryId(Guid.Parse(valueString))
```

Comment:
this hopes parse will not throw, can we TryParse and just continue without the id if we got a bad value? The id could come from other SDKs for example